spec/features/signing_up_spec.rb

```diff
@@ -0,0 +1,60 @@
+require_relative '../web_helpers'
+
+feature 'Signing up:' do
+  include Helpers
+
+  scenario 'User can sign up' do
+    visit '/users/new'
+    expect(page.status_code).to eq(200)
+    expect {sign_up}.to change(User, :count).by 1
+    expect(page).to have_content('Hi, Yev Insomniak')
+    expect(User.first.email).to eq('incredible@yev.com')
+  end
+
+  scenario 'Requires a matching confirmation password' do
+    expect {sign_up(password_confirmation: 'wrong')}.not_to change(User, :count)
+    expect(current_path).to eq ('/users')
+    expect(page).to have_content 'Password does not match the confirmation'
+  end
+
+  scenario "User can't sign up without an email address" do
+    visit '/users/new'
+    expect(page.status_code).to eq(200)
+    fill_in :name, with: 'Yev Insomniak'
+    fill_in :username, with: 'yev_insomniak'
+    fill_in :password, with: 'bla-bla-bla'
+    fill_in :password_confirmation, with: 'bla-bla-bla'
+    click_button 'Sign up!'
+    expect(page).to have_content('Email must not be blank')
+  end
+
+  scenario "User can't sign up without a valid email address" do
+    visit '/users/new'
+    expect(page.status_code).to eq(200)
+    fill_in :name, with: 'Yev Insomniak'
+    fill_in :username, with: 'yev_insomniak'
+    fill_in :email, with: 'incredibleyev.com'
+    fill_in :password, with: 'bla-bla-bla'
+    fill_in :password_confirmation, with: 'bla-bla-bla'
+    click_button 'Sign up!'
+    expect(page).to have_content("Doesn't look like an email address to me")
```

Comment:
If you were using factorygirl this would be a lot simpler and less brittle - `fill_in :name, with: user.name` would mean your tests are much more future proof.